### ISSUE TITLE
Doing inference with trained model

### ISSUE THREAD

**Issue by ghazalehtrb** [Anirudh257/strm#6]
Hi, Thank you for sharing the project codes!

I have trained a model on my costume dataset.
How can I modify the script to do inference? For example given a sample input, I want the model to give me the most probable class among all the support classes.

**Comment by Anirudh257** [comment]
Hi, I am glad that you found it helpful. 

For a custom dataset, you will need to follow the same steps as given in the paper for preprocessing, i.e.

Getting the 256 height rescaled images from the videos, etc.

For inference, you will need to make a lot of changes to the https://github.com/Anirudh257/strm/blob/main/video_reader.py code. The procedure can be:

1) Run the code with my data first and especially check the working of the video_reader.

2) See how the videos for each class are being loaded randomly here: https://github.com/Anirudh257/strm/blob/d74cd60ff81ffce6c21e248f2a7870a63e84df1e/video_reader.py#L246 .

3) From https://github.com/Anirudh257/strm/blob/d74cd60ff81ffce6c21e248f2a7870a63e84df1e/video_reader.py#L291, return the random indexes as well. 

4) Use https://github.com/Anirudh257/strm/blob/d74cd60ff81ffce6c21e248f2a7870a63e84df1e/run.py#L262 to get the most probable logits and use argmax to get the index of the top probable class. Using **indexes** from getitem, you can use it to get the name of the classes.

5) You can store some dict between the class names and indexes. These will be changing in every episode. So, modify the video_reader to return the indices and the dict.

I hope that this helps. Let me know if you have any other doubts.
